arlo_server/contests.py

```diff
@@ -107,13 +125,46 @@ def validate_contests(contests: List[dict]) -> None:
             )
 
 
+def round_status_by_contest(
+    round: Optional[Round], contests: List[Contest]
+) -> Dict[str, Optional[JSONDict]]:
+    if not round:
+        return {c.id: None for c in contests}
+
+    sampled_ballot_count_by_contest = dict(
+        SampledBallotDraw.query.filter_by(round_id=round.id)
+        .join(Batch)
+        .join(Jurisdiction)
+        .join(Jurisdiction.contests)
+        .group_by(Contest.id)
+        .values(Contest.id, func.count(SampledBallotDraw.batch_id))
+    )
```

Comment:
Would appreciate a sanity check on this. I'm still new to SQLAlchemy join logic.

Reply:
I find it helpful to plug queries into the python REPL. Often I get back the SQL if I omit the `.all()` or `.one()` part. With this one I have to omit `.values(…)`, which doesn't give you the exact query but it's close. Here's the actual query:

```sql
SELECT contest.id AS contest_id, count(sampled_ballot_draw.batch_id) AS count_1 
FROM sampled_ballot_draw JOIN batch ON batch.id = sampled_ballot_draw.batch_id JOIN jurisdiction ON jurisdiction.id = batch.jurisdiction_id JOIN contest_jurisdiction AS contest_jurisdiction_1 ON jurisdiction.id = contest_jurisdiction_1.jurisdiction_id JOIN contest ON contest.id = contest_jurisdiction_1.contest_id 
WHERE sampled_ballot_draw.round_id = %(round_id_1)s GROUP BY contest.id
```

I only managed to get that by substituting in an invalid value (the int `1`) for `round_id`, which then showed me the query in an exception.

Reply:
what is your goal with `func.count(SampledBallotDraw.batch_id)` as opposed to `func.count()`? Are you trying to count distinct batches? If so, that's not going to work, you'd need to `count(distinct(..))` or also group by the `batch_id`.

But I don't think that's what you're trying to do, so I imagine `func.count()` is all you want. I think that's the only tweak I've made, and as best as I can tell the query is right.

Reply:
Oooh cool trick!

Reply:
@benadida I just didn't know you could do that. I assumed you had to count a column in particular.